### PR TITLE
feat: useRecommendProfileToggle hook

### PR DESCRIPTION
**Pull request by krzysu**

### CODE REVIEW

File: packages/react/src/profile/infrastructure/ProfileCacheManager.ts [hunk 1]
```diff
@@ -49,6 +50,33 @@ export class ProfileCacheManager implements IProfileCacheManager {
     });
   }
 
+  update(
+    profileId: ProfileId,
+    updateFn: <TProfile extends Profile>(current: TProfile) => TProfile,
+  ) {
+    this.client.cache.updateQuery<ProfileData, ProfileVariables>(
```
Comment (krzysu) [inline packages/react/src/profile/infrastructure/ProfileCacheManager.ts:57]: @cesarenaldi I had to introduce cache update in ProfileCacheManager to optimistically flip the new peerToPeerRecommendedByMe flag, but it doesn't work (again) :(
the apollo error says "Missing field 'result' while writing result {}" what could it be?

File: packages/domain/src/use-cases/publications/ToggleProperty.ts [hunk 2]
```diff
@@ -1,43 +1,41 @@
-import { PublicationId } from '../../entities';
-
-export type TogglePropertyRequest = {
-  publicationId: PublicationId;
+export type TogglePropertyRequest<I> = {
+  id: I;
```
Comment (cesarenaldi) [inline packages/domain/src/use-cases/publications/ToggleProperty.ts:2]: If you make the `id: string` you won't need `I` generic, which simplifies a lot of changes in this PR.

Comment (krzysu) [reply packages/domain/src/use-cases/publications/ToggleProperty.ts]: done

File: packages/react/src/profile/infrastructure/ProfileCacheManager.ts (hunk 1 above)
Comment (krzysu) [reply packages/react/src/profile/infrastructure/ProfileCacheManager.ts]: replaced with `updateFragment`

**Pull request merged**